src/Indicator.vala

```diff
@@ -210,18 +210,14 @@ public class DateTime.Indicator : Wingpanel.Indicator {
 
             event_grid.add (menuitem);
 
-            /* Color events per calendar
-             * XXX: Day counter of trying this without help: 5 (Remove this when successful)
-             */
-            foreach (E.Source source in e.sources) {
-                var cal = (E.SourceCalendar)source.get_extension (E.SOURCE_EXTENSION_CALENDAR);
+            /* Color menuitem per calendar source of event */
+            var css_class = Util.get_style_calendar_color (e.cal);
+            menuitem.get_style_context ().add_class (css_class);
+            menuitem_icon.get_style_context ().add_class (css_class);
```

Comment:
I think there's no need for this line, because `menuitem_icon` is inside `menuitem` so the coloring is dealt to both.